### PR TITLE
UnsyncedBatchNorm with synced trainable weights for hlb cifar

### PR DESCRIPTION
**Pull request by chaosagent**
Developed for #3361. Previous UnsyncedBatchNorm kept separate weights and running statistics for every GPU, but weights are learnable parameters and should be optimized against the entire global batch. In the old UnsyncedBatchNorm, this would result in 1/6th the LR for weights for 6 GPUs, and also desynced learnable parameters.

Implemented ad-hoc in hlb_cifar10.py right now, but I think this can replace nn.BatchNorm2d as-is (this should work even without MLB, need to try)

Results in ~0.5-1.0% accuracy gain on CIFAR with no noticeable slowdown (maybe it's actually faster). Also removes the Tensor.stack kernels needed from old UnsyncedBatchNorm

Am benchmarking...

Old:
```
LATEBEAM=2 HALF=1 STEPS=350 BS=1536 GPUS=6
348   44.00 ms run,   26.53 ms python,   17.47 ms HIP * 6,  503.00 loss, 0.000099 LR, 6.76 GB used,  46090.57 GFLOPS,   2027.99 GOPS
349   43.81 ms run,   26.95 ms python,   16.86 ms HIP * 6,  501.50 loss, 0.000086 LR, 6.76 GB used,  46292.03 GFLOPS,   2027.99 GOPS
shuffling test dataset in 43.00 ms (epoch=0)
eval     10009/10752 93.09%,    0.43 val_loss STEP=350 (in 2164.65 ms)

real    0m50.613s

LATEBEAM=2 HALF=1 STEPS=350 BS=3072 GPUS=6
348   62.89 ms run,   30.54 ms python,   32.34 ms HIP * 6,  494.25 loss, 0.000099 LR, 12.67 GB used,  64486.94 GFLOPS,   4055.33 GOPS
349   62.76 ms run,   30.47 ms python,   32.29 ms HIP * 6,  493.50 loss, 0.000086 LR, 12.67 GB used,  64620.31 GFLOPS,   4055.33 GOPS
shuffling test dataset in 43.85 ms (epoch=0)
eval     11507/12288 93.64%,    0.44 val_loss STEP=350 (in 2118.33 ms)

real    1m7.084s
```

New:
```
LATEBEAM=2 HALF=1 STEPS=350 BS=1536 GPUS=6
348   43.61 ms run,   28.80 ms python,   14.81 ms HIP * 6,  494.25 loss, 0.000099 LR, 6.76 GB used,  46445.77 GFLOPS,   2025.50 GOPS
349   43.54 ms run,   29.40 ms python,   14.14 ms HIP * 6,  490.50 loss, 0.000086 LR, 6.76 GB used,  46518.03 GFLOPS,   2025.50 GOPS
shuffling test dataset in 42.74 ms (epoch=0)
eval     10098/10752 93.92%,    0.42 val_loss STEP=350 (in 1392.27 ms)

real    0m48.222s

LATEBEAM=2 HALF=1 STEPS=250 BS=3072 GPUS=6
248   63.10 ms run,   32.29 ms python,   30.80 ms HIP * 6,  493.00 loss, 0.000102 LR, 12.67 GB used,  64197.95 GFLOPS,   4050.60 GOPS
249   63.07 ms run,   32.94 ms python,   30.14 ms HIP * 6,  493.00 loss, 0.000086 LR, 12.67 GB used,  64219.31 GFLOPS,   4050.60 GOPS
shuffling test dataset in 43.06 ms (epoch=0)
eval     11499/12288 93.58%,    0.43 val_loss STEP=250 (in 1331.87 ms)

real    0m53.283s

LATEBEAM=2 HALF=1 STEPS=350 BS=3072 GPUS=6
348   62.78 ms run,   32.53 ms python,   30.25 ms HIP * 6,  477.75 loss, 0.000099 LR, 12.67 GB used,  64515.64 GFLOPS,   4050.60 GOPS
349   62.85 ms run,   32.32 ms python,   30.52 ms HIP * 6,  478.25 loss, 0.000086 LR, 12.67 GB used,  64452.23 GFLOPS,   4050.60 GOPS
shuffling test dataset in 43.62 ms (epoch=0)
eval     11562/12288 94.09%,    0.43 val_loss STEP=350 (in 1332.15 ms)

real    1m4.951s


```

Dealing with shape 1 on axis is tricky. Maybe make len(self.real_lbs)==1 => axis = None? But then you lose the implicit axis transformd originating from that buffer

### CODE REVIEW

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -927,11 +927,11 @@ def layernorm(self, axis=-1, eps:float=1e-5) -> Tensor:
     y = (self - self.mean(axis, keepdim=True))
     return y.mul((y*y).mean(axis, keepdim=True).add(eps).rsqrt())
 
-  def batchnorm(self, weight:Optional[Tensor], bias:Optional[Tensor], mean:Tensor, invstd:Tensor) -> Tensor:
-    shape = (1, -1) + (1,) * (self.ndim-2)
+  def batchnorm(self, weight:Optional[Tensor], bias:Optional[Tensor], mean:Tensor, invstd:Tensor, axis:Union[int,Tuple[int,...]]=1) -> Tensor:
+    shape = tuple(s if ax in argfix(axis) else 1 for ax, s in enumerate(self.shape))
```
Comment (chenyuxyz) [inline tinygrad/tensor.py:931]: Do `argfix(axis)` first so it's not being called multiple time. Also can you add a direct unit test for case when axis is a tuple in test_ops?

**Review by chenyuxyz — COMMENTED** [review]
Tested with 1, 2, 6 gpus and sync/unsynced and all looks good. One comment

**Comment by chenyuxyz** [comment]
tried with a few different seeds and for 6 GPU the eval acc improved by ~0.5%
